htdocs/includes/db_lib.php

```diff
@@ -7155,6 +7177,41 @@ function get_all_satellite_lab_ids()
 	return $satellite_lab_ids;
 }
 
+
+function get_all_satellite_lab_names()
+{
+    global $con, $LIS_SATELLITE_LAB_USER;
+
+    # Retrieves all user IDs corresponding to $LIS_SATELLITE_LAB_USER from the user table
+    $saved_db = DbUtil::switchToGlobal();
+	$query_string = "SELECT satellite_lab_name FROM user WHERE level = $LIS_SATELLITE_LAB_USER";
+	$resultset = query_associative_all($query_string);
+
+	$satellite_lab_names = array();
+	if (count($resultset) > 0)
+	{
+		foreach($resultset as $record)
+		{
+			if (isset($record["satellite_lab_name"])){
+				$satellite_lab_names[] = $record["satellite_lab_name"];
+			}
+		}
+	}
+	return $satellite_lab_names;
+}
+
+function get_satellite_lab_id_by_name($satellite_lab_name)
+{
+	global $con, $LIS_SATELLITE_LAB_USER;
+
+	# Retrieves all user IDs corresponding to $LIS_SATELLITE_LAB_USER from the user table
+	$saved_db = DbUtil::switchToGlobal();
+	$query_string = "SELECT satellite_lab_id FROM user WHERE satellite_lab_name = '$satellite_lab_name'";
```

Comment:
```suggestion
	$query_string = "SELECT satellite_lab_id FROM user WHERE satellite_lab_name = '$satellite_lab_name' LIMIT 1";
```

`query_associative_one` I don't believe enforces the query only returns one thing.

Reply:
This has been addressed in latest iteration.